openforcefield/typing/engines/smirnoff/forcefield.py

```diff
@@ -817,11 +817,33 @@ def _load_smirnoff_data(self, smirnoff_data, allow_cosmetic_attributes=False):
             Whether to permit non-spec kwargs in smirnoff_data.
         """
         import packaging.version
+
+        # Check that the SMIRNOFF version of this data structure is supported by this ForceField implementation
+
+        if "SMIRNOFF" in smirnoff_data.keys():
+            version = smirnoff_data['SMIRNOFF']['version']
+        elif "SMIRFF" in smirnoff_data.keys():
+            version = smirnoff_data['SMIRFF']['version']
+        else:
+            raise ParseError("'version' attribute must be specified in SMIRNOFF tag")
+
+        self._check_smirnoff_version_compatibility(str(version))
+        # Convert 0.1 spec files to 0.3 SMIRNOFF data format by converting
+        # from 0.1 spec to 0.2, then 0.2 to 0.3
+        if packaging.version.parse(str(version)) == packaging.version.parse("0.1"):
+            # NOTE: This will convert the top-level "SMIRFF" tag to "SMIRNOFF"
+            smirnoff_data = convert_0_1_smirnoff_to_0_2(smirnoff_data)
+            smirnoff_data = convert_0_2_smirnoff_to_0_3(smirnoff_data)
+
+        # Convert 0.2 spec files to 0.3 SMIRNOFF data format by removing units
+        # from section headers and adding them to quantity strings at all levels.
+        elif packaging.version.parse(str(version)) == packaging.version.parse("0.2"):
+            smirnoff_data = convert_0_2_smirnoff_to_0_3(smirnoff_data)
+
         # Ensure that SMIRNOFF is a top-level key of the dict
         if not('SMIRNOFF' in smirnoff_data):
             raise ParseError("'SMIRNOFF' must be a top-level key in the SMIRNOFF object model")
```

Comment:
The tests pass so I might be missing something, but I would expect this exception to be raised the `SMIRFF in smirnoff_data` instead of `SMIRNOFF`.

If this is a problem, it might be a good idea to substitute the `SMIRFF` key with `SMIRNOFF` in `smirnoff_data` above to avoid using a lot of if-else statement downstream.

Reply:
I'll make this clearer in the code, but the SMIRFF tag is converted to SMIRNOFF above (in `convert_0_1_smirnoff_to_0_2`)